File: Codechef/C Language/Covid_Run.c
#include<stdio.h>
int main()
{
	int t;
	scanf("%d", &t);
	while(t--)
	{
		int n, k, x, y, flag = 0;
		scanf("%d%d%d%d", &n, &k, &x, &y);
		for (int i = 1; i <= n; i++)
		{
			if((x+(k*i))%n == y)
			{
				printf("YES\n");
				flag = 1;
				break;
			}
		}
		if (flag == 0)
		{
			printf("NO\n");
		}
		
	}
	return 0;
}

File: Codechef/Python/Covid_Run.py
t = int(input())
for i in range(t):
	n, k, x, y = map(int,input().split())
	flag = 0;
	for i in range(n):
		if((x+(k*i))%n == y):
			print("YES")
			flag = 1
			break
	if flag == 0 :
		print("NO")